cfy_manager/components/postgresql_server/postgresql_server.py

```diff
@@ -216,29 +233,174 @@ def _update_postgres_password(self):
+        # We need access to the certs, which by default we don't have
+        common.sudo(['chmod', 'a+x', '/var/lib/patroni'])
+        # We currently use the same certificates for etcd, patroni,
+        # and postgres. This should be a reasonable starting approach as
+        # these reside on the same machine and all have the same impact if
+        # compromised (full access to data directly or via injected
+        # configuration changes).
+        for cert_deploy_args in (
+            {
+                'cert_destination': ETCD_SERVER_CERT_PATH,
+                'key_destination': ETCD_SERVER_KEY_PATH,
+                'ca_destination': ETCD_CA_PATH,
+                'owner': ETCD_USER,
+                'group': ETCD_GROUP,
+                'key_perms': '400',
+            },
+            {
+                'cert_destination': PATRONI_REST_CERT_PATH,
+                'key_destination': PATRONI_REST_KEY_PATH,
+                'owner': POSTGRES_USER,
+                'group': POSTGRES_GROUP,
+                'key_perms': '400',
+            },
+            {
+                'cert_destination': PATRONI_DB_CERT_PATH,
+                'key_destination': PATRONI_DB_KEY_PATH,
+                'ca_destination': PATRONI_DB_CA_PATH,
+                'owner': POSTGRES_USER,
+                'group': POSTGRES_GROUP,
+                'key_perms': '400',
+            },
+        ):
+            self.use_supplied_certificates(**cert_deploy_args)
+        common.sudo(['chmod', 'a-x', '/var/lib/patroni'])
+
+        logger.info('Deploying cluster config files.')
+        files.deploy(os.path.join(CONFIG_PATH, 'patroni.conf'),
+                     PATRONI_CONFIG_PATH)
+        common.sudo(['chown', 'root.postgres', PATRONI_CONFIG_PATH])
+        common.sudo(['chmod', '640', PATRONI_CONFIG_PATH])
+        files.deploy(os.path.join(CONFIG_PATH, 'etcd.conf'), ETCD_CONFIG_PATH)
+        common.sudo(['chown', 'postgres.', '/var/lib/patroni'])
+        common.sudo(['chmod', '700', '/var/lib/patroni'])
+        common.sudo(['chmod', '700', '/var/lib/patroni/data'])
+
+        logger.info('Configuring etcd')
+        systemd.enable('etcd', append_prefix=False)
+        self._start_etcd()
+        etcdctl_base_command = [
+            'etcdctl', '--endpoints', 'https://127.0.0.1:2379',
+            '--ca-file', ETCD_CA_PATH,
+        ]
+        cluster_state = common.run(
+            etcdctl_base_command + ['cluster-health'],
+            ignore_failures=True,
+        ).aggr_stdout
+        if 'cluster is degraded' in cluster_state:
+            cluster_config = config[POSTGRESQL_SERVER]['cluster']
+            # We want to configure etcd auth when the second node joins
+            # because that's the earliest we can do so.
+            logger.info('Configuring etcd authentication')
+            # Note that, per the etcd documentation, the root user must exist
+            # if authentication is to be enabled for etcd
+            common.run(
```

Comment:
common.run has a stdin kwarg, so the `echo | ` is likely unnecessary

Reply:
Good point. Updated.